setup.py

```diff
@@ -59,57 +58,54 @@ def run(self, *arg, **kw):
 with open('README.rst') as f:
     readme = f.read()
 
-install_reqs = [
+installReqs = [
     'bcrypt',
     'boto3',
     # CherryPy version is restricted due to a bug in versions >=11.1
     # https://github.com/cherrypy/cherrypy/issues/1662
     'CherryPy<11.1',
     'click',
     'filelock',
+    'funcsigs ; python_version < \'3.5\'',
     'jsonschema',
     'Mako',
     'pymongo>=3.5',
     'PyYAML',
-    'requests',
     'psutil',
     'python-dateutil',
     'pytz',
-    'six>=1.9'
+    'requests',
+    'shutilwhich ; python_version < \'3.3\'',
+    'six>=1.9',
 ]
 
-extras_reqs = {
-    'celery_jobs': ['celery'],
-    'dicom_viewer': ['pydicom'],
-    'geospatial': ['geojson'],
-    'item_tasks': ['ctk-cli'],
-    'ldap': ['pyldap'],
-    'thumbnails': ['Pillow', 'pydicom', 'numpy'],
-    'worker': ['celery>=4.0.0']
-}
-all_extra_reqs = itertools.chain.from_iterable(extras_reqs.values())
-extras_reqs['plugins'] = list(set(all_extra_reqs))
+extrasReqs = {}
+# To avoid conflict with the `girder-install plugin' command, this only adds built-in plugins with
+# extras requirements.
+with open(os.path.join('plugins', '.gitignore')) as builtinPluginsIgnoreStream:
+    builtinPlugins = set()
+    for line in builtinPluginsIgnoreStream:
+        # Plugin .gitignore entries should end with a /, but we will tolerate those that don't;
+        # (accordingly, note the non-greedy qualifier for the match group)
+        builtinPluginNameRe = re.match(r'^!(.+?)/?$', line)
+        if builtinPluginNameRe:
```

Comment:
nitpick: append to the list only if directory exists

Reply:
`builtinPlugins` is only an intermediate set, to be used in the following loop, so it will indeed contain false positive entries like `.gitignore`. However, we only query `builtinPlugins` for real directories that are returned by `os.listdir('plugins')`.

If someone else is planning on using `builtinPlugins` for another purpose in `setup.py`, then we should indeed perform additional validation and filtering. However, as the most recent comment indicates, this code is only a Girder-specific temporary measure to hold us over until pip-installable plugins force each plugin to have its own `setup.py` (instead of its own `requirements.txt`).

@jcfr With this context, do you think it's safe to keep the code simple as it is now, or should we add further guards against future people abusing this variable for other purposes?

Reply:
Thanks for the clarification. 

> do you think it's safe to keep the code simple as it is now

👍 

May be could add 

```
del builtinPlugins
```

of declare a an internal function named `def _get_directories_from_gitignore()`

and set:

```
builtinPlugins =  _get_directory_names_from_gitignore()
```

after the loop.

Reply:
@jcfr I've pushed another commit to ensure `builtinPlugins` is accurate. Do you think this is adequate?